program/src/state/rules.rs

```diff
@@ -65,12 +65,18 @@ pub enum Rule {
         /// The field in the `Payload` to be compared.
         field: String,
     },
-    /// A resulting derivation of seeds must match to a `Pubkey`.
-    DerivedKeyMatch {
-        /// The `Pubkey` to be compared against.
-        account: Pubkey,
-        /// The field in the `Payload` to be compared.
-        field: String,
+    /// A resulting PDA derivation of seeds must prove
+    /// the account is a PDA.
+    PDAMatch {
+        /// The program used for the PDA derivation.  If
+        /// `None` then the account owner is used.
+        program: Option<Pubkey>,
+        /// The field in the `Payload` to be compared
+        /// when looking for the PDA.
+        pda_field: String,
+        /// The field in the `Payload` to be compared
+        /// when looking for the seeds.
+        seeds_field: String,
```

Comment:
I'm wondering if this should be an array for PDAs like AH trade states that use lots of fields.

Reply:
I can review that.  If its an array then do we need to know how to match them up to the PayloadType values they are meant to key to?  In general I still feel like its unclear to the user what needs to be provided in the Payload and I can't think of anything better than adding comments to each rule that say what needs to be provided in the Payload.